bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,199 @@
+from asyncio import TimeoutError
+from random import choice
+from typing import Literal
+
+from discord import Embed, Message
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours, NEGATIVE_REPLIES
+
+# Defining all words in the list of words as a global variable
+with open("bot/resources/fun/hangman_words.txt", encoding="utf-8") as f:
+    ALL_WORDS = [line.strip('\n') for line in f.readlines()]
+
+# Defining a list of images that will be used for the game to represent the hangman person
+IMAGES = {
+    6: "https://cdn.discordapp.com/attachments/859123972884922418/888133201497837598/hangman0.png",
+    5: "https://cdn.discordapp.com/attachments/859123972884922418/888133595259084800/hangman1.png",
+    4: "https://cdn.discordapp.com/attachments/859123972884922418/888134194474139688/hangman2.png",
+    3: "https://cdn.discordapp.com/attachments/859123972884922418/888133758069395466/hangman3.png",
+    2: "https://cdn.discordapp.com/attachments/859123972884922418/888133786724859924/hangman4.png",
+    1: "https://cdn.discordapp.com/attachments/859123972884922418/888133828831477791/hangman5.png",
+    0: "https://cdn.discordapp.com/attachments/859123972884922418/888133845449338910/hangman6.png",
+}
+
+
+class Hangman(commands.Cog):
+    """
+    Cog for the Hangman game.
+
+    Hangman is a classic game where the user tries to guess a word, with a limited amount of tries.
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+
+    @staticmethod
+    def create_embed(tries: int, user_guess: str) -> Embed:
+        """
+        Helper method that creates the embed where the game information is shown.
+
+        This includes how many letters the user has guessed so far, and the hangman photo itself.
+        """
+        hangman_embed = Embed(
+            title="Hangman",
+            color=Colours.python_blue,
+        )
+        hangman_embed.set_image(url=IMAGES[tries])
+        hangman_embed.add_field(
+            name=f"You've guessed `{user_guess}` so far.",
+            value="Guess the word by sending a message with the letter!",
+            inline=False,
+        )
+        hangman_embed.set_footer(text=f"Tries: {tries}")
```

Comment:
```suggestion
        hangman_embed.set_footer(text=f"Tries remaining: {tries}")
```
My last comment,  I was confused while playing whether this meant the number of tries I've taken, or the number of tries I have left. I think this change will help the user understand what this number means.